### PR TITLE
DOC: Updating capitalization in folder doc/source/reference

### PR DESCRIPTION
**Pull request by cleconte987**
- [X] xref #32550

### CODE REVIEW

**Comment by WillAyd** [comment]
@datapythonista

**Review by datapythonista — APPROVED** [review]
Thanks for working on this @cleconte987, looks great.

What would be very useful is to add the files that are fixed to https://github.com/pandas-dev/pandas/blob/master/ci/code_checks.sh#L330, so no more errors are being introduced in those fails (the CI will fail and let us know if any PR introduces titles capitalized incorrectly).

If you can add it to this PR that would be great.

**Comment by cleconte987** [comment]
Ok I will work on that!

**Comment by datapythonista** [comment]
Thanks @cleconte987, changes look great. But looks like there are still several errors in that directory. Most of them need to be added here: https://github.com/pandas-dev/pandas/blob/master/scripts/validate_rst_title_capitalization.py#L20

Do you mind adding them, and fixing the cases that need it, if any? Thanks!

File: doc/source/reference/series.rst [hunk 1]
```diff
@@ -110,7 +110,7 @@ Binary operator functions
    Series.product
    Series.dot
 
-Function application, groupby & window
+Function application, GroupBy & window
```
Comment (datapythonista) [inline doc/source/reference/series.rst]: What's the reason for this one? Just my opinion, but I think this one was correct, and we don't want to capitalize `groupy` to `GroupBy`.

**Review by datapythonista — COMMENTED** [review]
Really nice, not sure about one of the cases, but other than that looks great.

File: doc/source/reference/series.rst (hunk 1 above)
Comment (cleconte987) [reply doc/source/reference/series.rst]: I might have wrongly seen it with capital letters or something, re-updating PR

**Comment by cleconte987** [comment]
Ok.. again failures in the PR request sorry for that. I know now why I putted "GroupBy" as an exception. Because it is written like that in title in file doc/source/reference/groupby.rst
I will re-update the PR. Should I add it to exceptions or not? I think it should

File: doc/source/reference/series.rst (hunk 1 above)
Comment (datapythonista) [reply doc/source/reference/series.rst]: I see your point. to me it feels that there is the class `GroupBy`, but here it's not talking about the class, but about the concept. I guess we can use `group by` or `groupping by` to disambiguate. But not that important, we can merge it like this and change it later if we think that makes sense.

**Review by datapythonista — APPROVED** [review]
Thanks, nice work.

**Comment by cleconte987** [comment]
Thank you.

Yes I understand now for the case of "groupby/GroupBy", but yeah it was not passing the tests then. Ok :)

**Comment by simonjayhawkins** [comment]
@cleconte987 can you merge upstream master to resolve conflicts

**Comment by datapythonista** [comment]
I guess everything that this PR was proposing has already been fixed in master, since no changes appear in the diff. Closing, let me know if that's not correct @cleconte987, happy to reopen this is the problem is another one.

**Comment by cleconte987** [comment]
No everything was updated, indeed!

**Pull request closed without merging**